### PR TITLE
update configs based on new label updates

### PR DESCRIPTION
**Pull request by josephperrott**
Update the ng-dev and  angular robot configurations to reflect the new label updates

### CODE REVIEW

**Review by devversion — APPROVED** [review]
LGTM. These labels were not updated as part of the new labeling doc, so they hadn't been updated. It makes sense to have them consistent with the target labels now though.

File: .ng-dev/merge.ts [hunk 1]
```diff
@@ -10,16 +10,17 @@ export const merge: DevInfraMergeConfig['merge'] = async api => {
...
     commitMessageFixupLabel: 'commit message fixup',
     labels: await getDefaultTargetLabelConfiguration(api, github, '@angular/core'),
     requiredBaseCommits: {
       // PRs that target either `master` or the patch branch, need to be rebased
       // on top of the latest commit message validation fix.
       // These SHAs are the commits that update the required license text in the header.
       'master': '5aeb9a4124922d8ac08eb73b8f322905a32b0b3a',
-      '10.0.x': '27b95ba64a5d99757f4042073fd1860e20e3ed24'
+      '10.0.x': '27b95ba64a5d99757f4042073fd1860e20e3ed24',
+      '10.1.x': 'fb0690323753fd683f4d546e063d6bce353847f7',
```
Comment (devversion) [inline .ng-dev/merge.ts:23]: Nit: Why did you add this for the `10.1.x` branch? I think we only want to set required base commits if we landed large refactoring's which PRs targeting this branch _definitely_ need to be based on (e.g. when we changed the formatter or codeowners)

Also if we ever add stuff here, a comment should be added as per above.

Comment (josephperrott) [reply .ng-dev/merge.ts:23]: I suppose that `10.1.x` will always include the necessary commits, but my thinking was just to establish the minimum commit for the branch.

I can remove it, It just seemed odd that one of the active branches isn't listed.

Comment (devversion) [reply .ng-dev/merge.ts:23]: I think there is no benefit in doing that. We should not have to bother doing that for all version branches as that will require unnecessary manual changes to the merge configuration on every branch-off. Only, for those version-branches being active and for which a minimum SHA is required due to relevant CI checks that could be hidden (that was the whole purpose of this check as per old merge script and new one), the config should be manually updated. Otherwise there seems no benefit.

Comment (josephperrott) [reply .ng-dev/merge.ts]: Removed.

**Review by devversion — APPROVED** [review]
